include/eve/detail/function/broadcast_group.hpp

```diff
@@ -0,0 +1,79 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/meta.hpp>
+#include <eve/detail/overload.hpp>
+#include <eve/concept/vectorized.hpp>
+#include <eve/pattern.hpp>
+#include <bit>
+
+namespace eve
+{
+  EVE_MAKE_CALLABLE(broadcast_group_, broadcast_group);
+
+  //================================================================================================
+  // Premade pattern generator
+  template<std::ptrdiff_t G, std::ptrdiff_t I, std::ptrdiff_t N>
+  inline constexpr auto broadcast_group_n = fix_pattern<N>( [](auto i, auto) { return I*G + i%G; } );
+
+  //================================================================================================
+  // Classify a pattern as a bcstg
+  namespace detail
+  {
+    // Find the proper broadcast group shape. IF none is found, returns {-1,-1}
+    template<std::size_t Size>
+    constexpr inline auto find_broadcast_group(std::array<std::ptrdiff_t,Size> const& pattern)
+    {
+      struct bg { std::ptrdiff_t group, index; };
+
+      for(std::ptrdiff_t g = Size;g > 0;g /= 2)
+      {
+        std::ptrdiff_t nb_idx = Size/g;
+
+        for(std::ptrdiff_t idx = 0;idx < nb_idx;idx++)
+        {
+          // Compute what we expect
+          std::ptrdiff_t expected[Size];
+          for(std::size_t i=0;i<Size;++i) expected[i] = idx*g+(i%g);
+
+          // Is it what we want ?
+          bool ok = true;
+          for(std::size_t i=0;i<Size;++i) ok = ok && (expected[i] == pattern[i]);
+
+          // Yes, stop the press! and exit!
+          if( ok ) return bg{g,idx};
+        }
+      }
+
+      return bg{-1,-1};
+    }
+
+    // Storage for group/index deduction
+    template<auto Group, auto Index> struct broadcast_group_status
+    {
+      static constexpr auto group = Group;
+      static constexpr auto index = Index;
+      constexpr operator bool() const { return group != -1; }
```

Comment:
Oh. Eh. Wait.
I can do two different broadcasts with the same Group and Index right?
And please explicit on the operator bool

Reply:
Added explicit